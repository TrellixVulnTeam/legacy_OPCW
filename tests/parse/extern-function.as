
# NOTE: "int" or "uint" should be replaced by something like
#       "std.libc.int" or "std.libc.size_t" so that the compiler will insert
#       the appropriate type for the target platform.

# Create an extern specification for the "exit" function.
extern let exit(int);

# Create an extern specification for the "socket" function.
extern let socket(domain: int, type_: int, protocol: int) -> int;

# Create an extern specification for the "recv" function.
extern let recv(socket: int, buffer: *uint8, length: uint, flags: int) -> int;
