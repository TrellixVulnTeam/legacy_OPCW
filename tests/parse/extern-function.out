ModuleDecl <?> _
  Nodes <?> 
    ExternFunc <?> exit
      Parameters <?> 
        FuncParam <?>
          Ident <?> int
    ExternFunc <?> socket
      Ident <?> int
      Parameters <?> 
        FuncParam <?> domain
          Ident <?> int
        FuncParam <?> type_
          Ident <?> int
        FuncParam <?> protocol
          Ident <?> int
    ExternFunc <?> recv
      Ident <?> int
      Parameters <?> 
        FuncParam <?> socket
          Ident <?> int
        FuncParam <?> buffer
          PointerType <?>
            Ident <?> uint8
        FuncParam <?> length
          Ident <?> uint
        FuncParam <?> flags
          Ident <?> int
